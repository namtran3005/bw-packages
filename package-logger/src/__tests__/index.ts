import { logger, loggers } from '../index';

describe('logger package', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });
  afterAll(() => {
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it('Should create new label', () => {
    logger('testing');
    expect(loggers.testing).toMatchSnapshot();
  });

  it('Should not create a new on existing labels', () => {
    logger('testing');
    const originalLabel = { ...loggers.testing };
    const returnedLabel = logger('testing');

    expect(returnedLabel).toEqual(originalLabel);
  });
});
