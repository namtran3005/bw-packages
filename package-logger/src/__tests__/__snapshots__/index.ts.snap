// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logger package Should create new label 1`] = `
DerivedLogger {
  "_events": Object {
    "data": [Function],
    "end": [Function],
    "prefinish": [Function],
  },
  "_eventsCount": 3,
  "_maxListeners": undefined,
  "_readableState": ReadableState {
    "autoDestroy": false,
    "awaitDrain": 0,
    "buffer": BufferList {
      "head": null,
      "length": 0,
      "tail": null,
    },
    "decoder": null,
    "defaultEncoding": "utf8",
    "destroyed": false,
    "emitClose": true,
    "emittedReadable": false,
    "encoding": null,
    "endEmitted": false,
    "ended": false,
    "flowing": true,
    "highWaterMark": 16,
    "length": 0,
    "needReadable": true,
    "objectMode": true,
    "paused": false,
    "pipes": Console {
      "__winstonerror": [Function],
      "__winstonwarn": [Function],
      "_events": Object {
        "close": [Function],
        "drain": [Function],
        "error": Array [
          [Function],
          [Function],
        ],
        "finish": [Function],
        "unpipe": Array [
          [Function],
          [Function],
        ],
        "warn": [Function],
      },
      "_eventsCount": 6,
      "_maxListeners": 30,
      "_writableState": WritableState {
        "bufferProcessing": false,
        "bufferedRequest": null,
        "bufferedRequestCount": 0,
        "corked": 0,
        "corkedRequestsFree": CorkedRequest {
          "entry": null,
          "finish": [Function],
          "next": null,
        },
        "decodeStrings": true,
        "defaultEncoding": "utf8",
        "destroyed": false,
        "ended": false,
        "ending": false,
        "errorEmitted": false,
        "finalCalled": false,
        "finished": false,
        "highWaterMark": 16,
        "lastBufferedRequest": null,
        "length": 0,
        "needDrain": false,
        "objectMode": true,
        "onwrite": [Function],
        "pendingcb": 0,
        "prefinished": false,
        "sync": true,
        "writecb": null,
        "writelen": 0,
        "writing": false,
      },
      "consoleWarnLevels": Object {},
      "eol": "
",
      "format": Format {
        "Format": [Function],
        "options": Object {},
      },
      "handleExceptions": undefined,
      "handleRejections": undefined,
      "level": "info",
      "levels": Object {
        "debug": 5,
        "error": 0,
        "http": 3,
        "info": 2,
        "silly": 6,
        "verbose": 4,
        "warn": 1,
      },
      "name": "console",
      "parent": [Circular],
      "silent": true,
      "stderrLevels": Object {},
      "writable": true,
      Symbol(kCapture): false,
    },
    "pipesCount": 1,
    "readableListening": false,
    "reading": false,
    "readingMore": false,
    "resumeScheduled": true,
    "sync": false,
  },
  "_transformState": Object {
    "afterTransform": [Function],
    "needTransform": false,
    "transforming": false,
    "writecb": null,
    "writechunk": null,
    "writeencoding": null,
  },
  "_writableState": WritableState {
    "autoDestroy": false,
    "bufferProcessing": false,
    "bufferedRequest": null,
    "bufferedRequestCount": 0,
    "corked": 0,
    "corkedRequestsFree": CorkedRequest {
      "entry": null,
      "finish": [Function],
      "next": null,
    },
    "decodeStrings": true,
    "defaultEncoding": "utf8",
    "destroyed": false,
    "emitClose": true,
    "ended": false,
    "ending": false,
    "errorEmitted": false,
    "finalCalled": false,
    "finished": false,
    "highWaterMark": 16,
    "lastBufferedRequest": null,
    "length": 0,
    "needDrain": false,
    "objectMode": true,
    "onwrite": [Function],
    "pendingcb": 0,
    "prefinished": false,
    "sync": true,
    "writecb": null,
    "writelen": 0,
    "writing": false,
  },
  "allowHalfOpen": true,
  "debug": [Function],
  "defaultMeta": null,
  "error": [Function],
  "exceptions": ExceptionHandler {
    "handlers": Map {},
    "logger": [Circular],
  },
  "exitOnError": true,
  "format": Format {
    "options": Object {
      "stack": true,
    },
  },
  "http": [Function],
  "info": [Function],
  "level": "info",
  "levels": Object {
    "debug": 5,
    "error": 0,
    "http": 3,
    "info": 2,
    "silly": 6,
    "verbose": 4,
    "warn": 1,
  },
  "profilers": Object {},
  "readable": true,
  "rejections": RejectionHandler {
    "handlers": Map {},
    "logger": [Circular],
  },
  "silent": undefined,
  "silly": [Function],
  "verbose": [Function],
  "warn": [Function],
  "writable": true,
  Symbol(kCapture): false,
}
`;
